ui/src/profiles/ProfileModal.tsx

```diff
@@ -107,36 +71,43 @@ export default function ProfileModal() {
       <div className="p-5 pt-14">
         <div className="text-lg font-bold">
           <ShipName name={ship} showAlias />
-          {contact.nickname ? (
+          {contact && contact.nickname ? (
             <ShipName name={ship} className="ml-2 text-gray-600" />
           ) : null}
           <PalIcon className="ml-2" ship={ship} />
         </div>
-        <ProfileBio bio={contact.bio} />
-        {contact.groups.length > 0 && (
+        {contact && <ProfileBio bio={contact.bio} />}
+        {contact && contact.groups.length > 0 && (
           <div className="mt-5">
             <h2 className="mb-3 font-semibold">Favorite Groups</h2>
             <FavoriteGroupGrid groupFlags={contact.groups} />
           </div>
         )}
       </div>
       <footer className="flex items-center py-4 px-6">
-        {pals.installed && pals.pals.outgoing[ship.slice(1)] ? (
-          <button
-            className="secondary-button ml-auto bg-red-100"
-            onClick={() => pals.removePal(ship.slice(1))}
-          >
-            Remove Pal
-          </button>
-        ) : (
-          <button
-            className="secondary-button ml-auto"
-            onClick={() => pals.addPal(ship.slice(1))}
-          >
-            Add Pal
-          </button>
-        )}
-        <button className="secondary-button ml-2" onClick={handleCopyClick}>
+        {pals.installed &&
+          ship !== window.our &&
+          (pals.pals.outgoing[ship.slice(1)] ? (
+            <button
+              className="secondary-button ml-auto bg-red-100 dark:bg-red dark:text-white"
+              onClick={() => pals.removePal(ship.slice(1))}
+            >
+              Remove Pal
+            </button>
+          ) : (
+            <button
+              className="secondary-button ml-auto"
+              onClick={() => pals.addPal(ship.slice(1))}
+            >
+              Add Pal
+            </button>
+          ))}
+        <button
+          className={`secondary-button ${
+            pals.installed && ship !== window.our ? 'ml-2' : 'ml-auto'
+          }`}
```

Comment:
let's remove this and all the margins on the buttons and use `justify-end space-x-2` on the footer, should be simpler